### ISSUE TITLE
Office format is failing on AVX2

### ISSUE THREAD

**Issue by claudioandre** [magnumripper/JohnTheRipper#2921]
I know it works on AVX. Confirmed on well for AVX2 and super for AVX.

```
Testing: Office, 2007/2010/2013 [SHA1 256/256 AVX2 8x / SHA512 256/256 AVX2 4x AES]... (2xOMP) FAILED (cmp_all(64))
```

**Comment by magnumripper** [comment]
This gotta be introduced by recent BE fixes

**Comment by jfoug** [comment]
Hmm, it worked on AVX2 (256 bit) just fine before I checked it in. i will check and see why things are broken.

**Comment by jfoug** [comment]
I also waited around before going over to kids house for thanksgiving dinner, to make sure the CI bots were happy with that patch.  I saw the green check before I left.   It should be trivial to get it fixed.   

I easily 'could' revert to the older logic. Actually, for non BE, it IS the original logic, so I am not sure the problem.

**Comment by jfoug** [comment]
Hmm,  it is office 2007.  that was working.  I will get it fixed.

**Comment by jfoug** [comment]
```
$ ../run/john -test=3 -form=office -cost=2013
Will run 8 OpenMP threads
Benchmarking: Office, 2007/2010/2013 [SHA1 256/256 AVX2 8x / SHA512 256/256 AVX2 4x AES]... (8xOMP) DONE
Speed for cost 1 (MS Office version) of 2013, cost 2 (iteration count) of 100000
Raw:    196 c/s real, 31.3 c/s virtual

$ ../run/john -test-full=3 -form=office -cost=2013
Will run 8 OpenMP threads
Testing: Office, 2007/2010/2013 [SHA1 256/256 AVX2 8x / SHA512 256/256 AVX2 4x AES]... (8xOMP) PASS
$ ../run/john -test-full=3 -form=office
Will run 8 OpenMP threads
Testing: Office, 2007/2010/2013 [SHA1 256/256 AVX2 8x / SHA512 256/256 AVX2 4x AES]... (8xOMP) FAILED (cmp_all(256))

$ ../run/john -test-full=3 -form=office -cost=2010
Will run 8 OpenMP threads
Testing: Office, 2007/2010/2013 [SHA1 256/256 AVX2 8x / SHA512 256/256 AVX2 4x AES]... (8xOMP) PASS
$ ../run/john -test-full=3 -form=office -cost=2007
Will run 8 OpenMP threads
Testing: Office, 2007/2010/2013 [SHA1 256/256 AVX2 8x / SHA512 256/256 AVX2 4x AES]... (8xOMP) FAILED (cmp_all(256))
```

**Comment by jfoug** [comment]
Hmm, this was the main change for the 32 bit formats (2007/2010)

```diff
	// Finally, append "block" (0) to H(n)
	// hashBuf = SHA1Hash(hashBuf, 0);
-	for (i = 0; i < SIMD_PARA_SHA1; ++i)
-		memset(&keys[GETPOS_1(23,i*SIMD_COEF_32)], 0, 4*SIMD_COEF_32);
+	for (i = 0; i < SIMD_PARA_SHA1; ++i) {
+		int j;
+		for (j = 20; j < 24; ++j) {
+			keys[GETPOS_1(j,i*SIMD_COEF_32)] = 0;
+			keys[GETPOS_1(j,(i*SIMD_COEF_32+1))] = 0;
+			keys[GETPOS_1(j,(i*SIMD_COEF_32+2))] = 0;
+			keys[GETPOS_1(j,(i*SIMD_COEF_32+3))] = 0;
+		}
+	}
```

I do not see the problem.  I will dig deeper.

Oh, I might see the problem!!!

**Comment by jfoug** [comment]
I think this is the proper loop, testing now.

```c
for (i = 0; i < SHA1_LOOP_CNT; ++i) {
		keys[GETPOS_1(20,i)] = 0;
		keys[GETPOS_1(21,i)] = 0;
		keys[GETPOS_1(22,i)] = 0;
		keys[GETPOS_1(23,i)] = 0;
	}
```

**Comment by jfoug** [comment]
Fixed b74e9e432
